### PR TITLE
host grabs address from Announce now

### PR DESCRIPTION
**Pull request by DavidVorick**

### CODE REVIEW

File: modules/host/announce.go [hunk 1]
```diff
@@ -54,11 +55,26 @@ func (h *Host) Announce() error {
...
+		uc, err := h.wallet.NextAddress()
+		if err != nil {
+			return err
+		}
+		h.UnlockHash = uc.UnlockHash()
+		err = h.save()
+		if err != nil {
+			return err
+		}
+	}
```
Comment (lukechampine) [inline modules/host/announce.go]: reorder

File: modules/host/negotiate.go [hunk 2]
```diff
@@ -153,6 +153,11 @@ func (h *Host) considerRevision(txn types.Transaction, obligation contractObliga
 // rpcUpload is an RPC that negotiates a file contract. Under the new scheme,
 // file contracts should not initially hold any data.
 func (h *Host) rpcUpload(conn net.Conn) error {
+	// Check that the host has grabbed an address from the wallet.
+	if h.UnlockHash == (types.UnlockHash{}) {
+		return errors.New("host has not yet been properly initialized - call 'announce' to finish initialization")
```
Comment (lukechampine) [inline modules/host/negotiate.go]: users won't see this error; maybe it should be printed?

File: modules/host/announce.go (hunk 1 above)
Comment (DavidVorick) [inline modules/host/announce.go]: I did, though it kept your comment for some reason

File: modules/host/negotiate.go (hunk 2 above)
Comment (DavidVorick) [inline modules/host/negotiate.go]: hmm, I guess so. I'll add it.

**Comment by DavidVorick** [comment]
ah crap missed the println lol

**Pull request merged**